code/CMakeLists.txt

```diff
@@ -0,0 +1,320 @@
+		elseif ("${ext}" STREQUAL ".c")
+			get_filename_component(basename ${srcfile} NAME_WE)
+			set(q3asm_outfile "${CMAKE_CURRENT_BINARY_DIR}/${basename}.asm")
+			add_custom_command(
+				OUTPUT ${q3asm_outfile}
+				COMMAND q3lcc
+				ARGS ${compileflags} -o "\"${q3asm_outfile}\"" "\"${CMAKE_CURRENT_SOURCE_DIR}/${srcfile}\""
+				DEPENDS q3lcc ${CMAKE_CURRENT_SOURCE_DIR}/${srcfile}
+			)
+			set_source_files_properties(${q3asm_outfile} PROPERTIES GENERATED TRUE)
+			list(APPEND QVM_SRCS ${q3asm_outfile})
+		endif()
+	endforeach()
+	add_dependencies(qvm qvm_${TARGET})
+
+	set(QVM_PATH ${GAME_NAME}/vm/${TARGET_SHORT}.qvm)
+	add_custom_command(
+		OUTPUT "${ENGINE_BINARY_DIR}/${QVM_PATH}"
+		COMMAND q3asm
+		ARGS -o "${ENGINE_BINARY_DIR}/${QVM_PATH}" ${QVM_SRCS}
+		DEPENDS q3asm ${QVM_SRCS}
+	)
+	add_custom_target(qvm_${TARGET} DEPENDS "${ENGINE_BINARY_DIR}/${QVM_PATH}")
+	add_dependencies(${TARGET} qvm_${TARGET})
+	set_source_files_properties(${QVM_PATH} PROPERTIES GENERATED TRUE)
+	set_source_files_properties(${QVM_PATH} PROPERTIES MACOSX_PACKAGE_LOCATION Resources/${QVM_PATH})
+endfunction()
+
+function(add_botlib TARGET)
+	set(BOTLIB_SRCS
+		${CODE_DIR}/botlib/be_aas_bspq3.c
+		${CODE_DIR}/botlib/be_aas_cluster.c
+		${CODE_DIR}/botlib/be_aas_debug.c
+		${CODE_DIR}/botlib/be_aas_entity.c
+		${CODE_DIR}/botlib/be_aas_file.c
+		${CODE_DIR}/botlib/be_aas_main.c
+		${CODE_DIR}/botlib/be_aas_move.c
+		${CODE_DIR}/botlib/be_aas_optimize.c
+		${CODE_DIR}/botlib/be_aas_reach.c
+		${CODE_DIR}/botlib/be_aas_route.c
+		${CODE_DIR}/botlib/be_aas_routealt.c
+		${CODE_DIR}/botlib/be_aas_sample.c
+		${CODE_DIR}/botlib/be_ai_char.c
+		${CODE_DIR}/botlib/be_ai_chat.c
+		${CODE_DIR}/botlib/be_ai_gen.c
+		${CODE_DIR}/botlib/be_ai_goal.c
+		${CODE_DIR}/botlib/be_ai_move.c
+		${CODE_DIR}/botlib/be_ai_weap.c
+		${CODE_DIR}/botlib/be_ai_weight.c
+		${CODE_DIR}/botlib/be_ea.c
+		${CODE_DIR}/botlib/be_interface.c
+		${CODE_DIR}/botlib/l_crc.c
+		${CODE_DIR}/botlib/l_libvar.c
+		${CODE_DIR}/botlib/l_log.c
+		${CODE_DIR}/botlib/l_memory.c
+		${CODE_DIR}/botlib/l_precomp.c
+		${CODE_DIR}/botlib/l_script.c
+		${CODE_DIR}/botlib/l_struct.c
+	)
+	target_sources(${TARGET} PRIVATE ${BOTLIB_SRCS})
+	#target_include_directories(${TARGET} PRIVATE ${CODE_DIR}/botlib)
+	foreach(_file ${BOTLIB_SRCS})
+		set_property(SOURCE ${_file} APPEND PROPERTY COMPILE_DEFINITIONS BOTLIB)
+		set_property(SOURCE ${_file} APPEND PROPERTY INCLUDE_DIRECTORIES ${CODE_DIR}/botlib)
+	endforeach()
+endfunction()
+
+function(add_asm TARGET)
+	if (MSVC)
+		enable_language(ASM_MASM)
+		set(ASM_SRCS)
+		if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+				list(APPEND ASM_SRCS ../asm/vm_x86_64.asm)
+		endif()
+		list(APPEND ASM_SRCS ../asm/snapvector.asm ../asm/ftola.asm)
+		foreach(_file ${ASM_SRCS})
+			if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+				set_property(SOURCE ${_file} APPEND PROPERTY COMPILE_DEFINITIONS idx64=1)
+			endif()
+			set_property(SOURCE ${_file} APPEND PROPERTY LANGUAGE ASM_MASM)
```

Comment:
`			set_source_files_properties(${_file} PROPERTIES COMPILE_FLAGS "/safeseh")`

is missing here